### PR TITLE
[SUCE-447] Loading author data

### PR DESCRIPTION
**Pull request by SamaneriVimala**
Last few commits load the author data for po files.

### CODE REVIEW

File: server/server/data_loader/po.py [hunk 1]
```diff
@@ -87,12 +88,19 @@ def process_dir(change_tracker, po_dir, info):
         data = extract_strings_from_po_file(po_file)
         uid = remove_leading_zeros(po_file.stem)
 
+        root_author = get_author(info['root_author'], authors)
```
Comment (gradam) [inline server/server/data_loader/po.py]: You can merge `get_author` and `get_author_short` functions into one and return tuple. eg:
`root_author, root_author_short = get_author_names(info['root_author'], authors)`
This way you don't have to iterate over authors list twice.

File: server/server/data_loader/po.py [hunk 2]
```diff
@@ -123,17 +133,23 @@ def process_dir(change_tracker, po_dir, info):
         }
 
     for sub_folder in po_dir.glob('*/'):
-        yield from process_dir(change_tracker, sub_folder, info=info)
+        yield from process_dir(change_tracker, sub_folder, authors, info=info)
+
+def get_author_short(author_uid, authors):
+    for item in authors:
+        if (item['uid'] == author_uid):
```
Comment (gradam) [inline server/server/data_loader/po.py]: no need for ()

File: server/server/data_loader/po.py [hunk 3]
```diff
@@ -123,17 +133,23 @@ def process_dir(change_tracker, po_dir, info):
...
+    return None
 
-def insert_or_update(collection, doc):
-    try:
-        collection.insert(doc)
-    except DocumentInsertError as e:
-        collection.update(doc)
+def get_author(author_uid, authors):
+    for item in authors:
+        if (item['uid'] == author_uid):
```
Comment (gradam) [inline server/server/data_loader/po.py]: no need for ()

File: server/server/data_loader/po.py (hunk 1 above)
Comment (SamaneriVimala) [reply server/server/data_loader/po.py]: Thanks. Done that also for some older code.

File: server/server/data_loader/textdata.py [hunk 4]
```diff
@@ -131,8 +131,9 @@ def process_lang_dir(self, lang_dir, authors, data_dir=None, files_to_process=No
                             next_uid = None
 
                 author = self._get_author(root, lang_uid, uid)
-                author_short = self._get_author_short(author, authors)
-                author_uid = self._get_author_uid(author, authors)
+                author_data = self._get_author_data(author, authors)
```
Comment (gradam) [inline server/server/data_loader/textdata.py]: you can use something called tuple unpacking. so the proper code is:
`author_uid, author_short = self._get_author_data(author, authors)`

File: server/server/data_loader/textdata.py [hunk 5]
```diff
@@ -206,21 +207,13 @@ def _get_author(self, root, lang_uid, uid):
...
         return None
 
-    def _get_author_short(self, author, authors):
+    def _get_author_data(self, author, authors):
 
         for item in authors:
-            if (item['long_name'] == author): 
-                return item['short_name']
+            if item['long_name'] == author: 
+                return (item['short_name'], item['uid'])
```
Comment (gradam) [inline server/server/data_loader/textdata.py]: no need for ()

File: server/server/data_loader/po.py [hunk 6]
```diff
@@ -123,17 +131,16 @@ def process_dir(change_tracker, po_dir, info):
...
 
-def insert_or_update(collection, doc):
-    try:
-        collection.insert(doc)
-    except DocumentInsertError as e:
-        collection.update(doc)
+def get_author(author_uid, authors):
+    for item in authors:
+        if item['uid'] == author_uid: 
+            return (item['long_name'], item['short_name'])
```
Comment (gradam) [inline server/server/data_loader/po.py]: no need for ()

**Pull request merged**